core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java

```diff
@@ -0,0 +1,185 @@
+        try {
+            assertEquals(BigDecimal.ZERO, notContains.evaluate().getNumberValue());
+        } catch (EvaluationException | ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testAfterDate(){
+        Expression afterDate = new Expression("AFTERDATE(\"01 01, 2023 00:00:00\")", config);
+        try {
+            assertEquals(BigDecimal.ONE, afterDate.evaluate().getNumberValue());
+        } catch (EvaluationException | ParseException e) {
+            e.printStackTrace();
+        }
+        Expression notAfterDate = new Expression("AFTERDATE(\"01 01, 2100 00:00:00\")", config);
+        try {
+            assertEquals(BigDecimal.ZERO, notAfterDate.evaluate().getNumberValue());
+        } catch (EvaluationException | ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testBeforeDate(){
+        Expression beforeDate = new Expression("BEFOREDATE(\"01 01, 2100 00:00:00\")", config);
+        try {
+            assertEquals(BigDecimal.ONE, beforeDate.evaluate().getNumberValue());
+        } catch (EvaluationException | ParseException e) {
+            e.printStackTrace();
+        }
+        Expression notBeforeDate = new Expression("BEFOREDATE(\"01 01, 2023 00:00:00\")", config);
+        try {
+            assertEquals(BigDecimal.ZERO, notBeforeDate.evaluate().getNumberValue());
+        } catch (EvaluationException | ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testCheckTickFunction(){
+        PlayerWrapper<?> playerWrapper = this.getFactory().createPlayer();
+        CheckTickFunction checkTickFunction = new CheckTickFunction();
+        assertFalse(checkTickFunction.function(playerWrapper));
+    }
+
+    @Test
+    public void testConditionFunction(){
+        PlayerWrapper<?> playerWrapper = this.getFactory().createPlayer();
+        ConditionFunction conditionFunction = new ConditionFunction();
+        conditionFunction.setData("STREQUAL(\"test\", \"test\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("STREQUALIGNORECASE(\"tEST\", \"test\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("STRSTARTSWITH(\"test\", \"te\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("STRENDSWITH(\"testend\", \"end\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("STRCONTAINS(\"test\", \"es\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("AFTERDATE(\"01 01, 2023 00:00:00\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("BEFOREDATE(\"01 01, 2100 00:00:00\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+    }
```

Comment:
Can these be separated into smaller units I.E: `testStringEqualsTrue` `testStringEqualsFalse` `testStringEqualsEmpty` etc? The goal here is to have each have one small unit for each so we can catch edge cases easier without having to dig through the code.

Reply:
Good point. I'll split them up.